clients/src/test/java/org/apache/kafka/common/metrics/stats/FrequenciesTest.java

```diff
@@ -102,58 +110,69 @@ public void testBooleanFrequencies() {
-        assertEquals(0.25, (Double) metric2.metricValue(), DELTA);
-        assertEquals(0.25, (Double) metric3.metricValue(), DELTA);
-        assertEquals(0.25, (Double) metric4.metricValue(), DELTA);
+        assertEquals(0.25, metricValueFor("1"), DELTA);
+        assertEquals(0.25, metricValueFor("2"), DELTA);
+        assertEquals(0.25, metricValueFor("3"), DELTA);
+        assertEquals(0.25, metricValueFor("4"), DELTA);
+    }
+
+    @Test
+    public void testWithMetricsStrategy2() {
+        Frequencies frequencies = new Frequencies(4, 1.0, 4.0, freq("1", 1.0),
+                freq("2", 2.0), freq("3", 3.0), freq("4", 4.0));
+        Sensor sensor = metrics.sensor("test", config);
+        sensor.add(frequencies);
+
+        // Record 100 events half-half between 1st and 2nd buckets
+        for (int i = 0; i < 100; ++i) {
+            frequencies.record(config, i % 2 + 1, time.milliseconds());
+        }
+        assertEquals(0.50, metricValueFor("1"), DELTA);
+        assertEquals(0.50, metricValueFor("2"), DELTA);
+        assertEquals(0.00, metricValueFor("3"), DELTA);
+        assertEquals(0.00, metricValueFor("4"), DELTA);
+    }
 
-        // Record 2 windows worth of values
-        for (int i = 0; i != 100; ++i) {
+    @Test
+    public void testWithMetricsStrategy3() {
+        Frequencies frequencies = new Frequencies(4, 1.0, 4.0, freq("1", 1.0),
+                freq("2", 2.0), freq("3", 3.0), freq("4", 4.0));
+        Sensor sensor = metrics.sensor("test", config);
+        sensor.add(frequencies);
+
+        // Record 50 events half-half between 1st and 2nd buckets
+        for (int i = 0; i < 50; ++i) {
             frequencies.record(config, i % 2 + 1, time.milliseconds());
         }
-        assertEquals(0.50, (Double) metric1.metricValue(), DELTA);
-        assertEquals(0.50, (Double) metric2.metricValue(), DELTA);
-        assertEquals(0.00, (Double) metric3.metricValue(), DELTA);
-        assertEquals(0.00, (Double) metric4.metricValue(), DELTA);
-
-        // Record 1 window worth of values to overlap with the last window
-        // that is half 1.0 and half 2.0
-        for (int i = 0; i != 50; ++i) {
+        // Record 50 events to 4th bucket
+        for (int i = 0; i < 50; ++i) {
             frequencies.record(config, 4.0, time.milliseconds());
         }
-        assertEquals(0.25, (Double) metric1.metricValue(), DELTA);
-        assertEquals(0.25, (Double) metric2.metricValue(), DELTA);
-        assertEquals(0.00, (Double) metric3.metricValue(), DELTA);
-        assertEquals(0.50, (Double) metric4.metricValue(), DELTA);
+        assertEquals(0.25, metricValueFor("1"), DELTA);
+        assertEquals(0.25, metricValueFor("2"), DELTA);
+        assertEquals(0.00, metricValueFor("3"), DELTA);
+        assertEquals(0.50, metricValueFor("4"), DELTA);
     }
 
-    protected MetricName name(String metricName) {
+    private MetricName name(String metricName) {
```

Comment:
name => toMetricName ?

Reply:
Tried to make now `metricName`, `freq` and `metricValue` consistent in naming, please check if it's ok